wemake_python_styleguide/visitors/ast/keywords.py

```diff
@@ -71,6 +74,11 @@ def _check_exception_type(self, node: ast.Raise) -> None:
                 BaseExceptionRaiseViolation(node, text=exception_name),
             )
 
+    def _check_bare_raise(self, node: ast.Raise) -> None:
+        parent = walk.get_closest_parent(node, ast.ExceptHandler)
+        if not parent and node.exc is None:
+            self.add_violation(BareRaiseViolation(node))
```

Comment:
NIP: to optimize this code we can first check if the given `node` is bare `raise` statement (it's quite rarely used compared to `raise SomeException()`, so mostly it should evaluate to `False`) and then check its parent:

```suggestion
        if node.exc is None:
            if not walk.get_closest_parent(node, ast.ExceptHandler):
                self.add_violation(BareRaiseViolation(node))
```